src/HttpClient.ts

```diff
@@ -70,6 +72,20 @@ export class HttpClient {
         }
     }
 
+    public static getHttpToHttpsProxyConfig(
+        proxyConfig: IProxyConfig | false | undefined
+    ): HttpsProxyAgent | undefined {
+        if (
+            !proxyConfig ||
+            !proxyConfig.host ||
+            !proxyConfig.port ||
+            (proxyConfig.protocol && proxyConfig.protocol.includes('https'))
+        ) {
+            return undefined;
+        }
+        return new HttpsProxyAgent(`http://${proxyConfig.host}:${proxyConfig.port}`);
```

Comment:
What happens if it is an https->https proxy?

Reply:
can't be, we checked just before if the proxy protocol is https return undefined